### PR TITLE
feat: add kakao OIDC

### PR DESCRIPTION
**Pull request by MiryangJung**
Adds support for OIDC login with Kakao via `signInWithIdToken()`.

Fixes:
- https://github.com/supabase/gotrue/issues/1358

### CODE REVIEW

File: internal/api/provider/oidc.go [hunk 1]
```diff
@@ -312,6 +314,44 @@ func parseAzureIDToken(token *oidc.IDToken) (*oidc.IDToken, *UserProvidedData, e
 	return token, &data, nil
 }
 
+type KakaoIDTokenClaims struct {
+	jwt.StandardClaims
+
+	Email         string `json:"email"`
+	EmailVerified bool   `json:"email_verified"`
```
Comment (hf) [inline internal/api/provider/oidc.go:321]: This is not listed here: https://developers.kakao.com/docs/latest/en/kakaologin/common#id-token

It appears that the email in Kakao is always verified?

Please also extract the `nickname` and all other custom claims too.

**Comment by hf** [comment]
Looks good, please address the comments on the ID token contents and I'll merge! Thanks!

**Comment by hf** [comment]
Can you link to docs on how the app authentication works? Do you need to setup different client IDs (audiences) like you need with Google and Apple?

File: internal/api/provider/oidc.go (hunk 1 above)
Comment (MiryangJung) [reply internal/api/provider/oidc.go:321]: Yes, I will check and fix it.

**Comment by MiryangJung** [comment]
> Can you link to docs on how the app authentication works? Do you need to setup different client IDs (audiences) like you need with Google and Apple?

Yes need to setup kakao js key like google, and need additional work as well. I will add it to docs.

File: internal/api/provider/oidc.go (hunk 1 above)
Comment (MiryangJung) [reply internal/api/provider/oidc.go]:
```
Email saved in Kakao Account.
Only returned is_email_valid is true.
```

https://developers.kakao.com/docs/latest/en/kakaologin/rest-api#get-id-token-info-response-body

**Comment by softmarshmallow** [comment]
Hi. can we get a eta on this?

**Comment by J0** [comment]
@softmarshmallow @MiryangJung

Thanks for your patience with us, do you mind [updating the docs](https://supabase.com/docs/guides/auth/social-login/auth-kakao) with the changes needed to get Kakao OIDC up and running? We should be able to review once the docs are completed - if there's an existing PR that we've missed feel free to link it here and we can review.

Let us know!

**Comment by MiryangJung** [comment]
> @softmarshmallow @MiryangJung
> 
> Thanks for your patience with us, do you mind [updating the docs](https://supabase.com/docs/guides/auth/social-login/auth-kakao) with the changes needed to get Kakao OIDC up and running? We should be able to review once the docs are completed - if there's an existing PR that we've missed feel free to link it here and we can review.
> 
> Let us know!

I'll update the document soon and get back to you.

**Comment by MiryangJung** [comment]
> @softmarshmallow @MiryangJung
> 
> Thanks for your patience with us, do you mind [updating the docs](https://supabase.com/docs/guides/auth/social-login/auth-kakao) with the changes needed to get Kakao OIDC up and running? We should be able to review once the docs are completed - if there's an existing PR that we've missed feel free to link it here and we can review.
> 
> Let us know!

I added a docs!

https://github.com/supabase/supabase/pull/21229

**Comment by J0** [comment]
Btw @MiryangJung @softmarshmallow, thank you so much for taking the time to write docs. Just to give you an update - one of our team members is testing the flow and once they are done we should be good to go.

I know it's probably tiring to wait but we seek your patience - we think this is a good feature to have but we want to ensure that there are no issues with the rollout

**Comment by MiryangJung** [comment]
> Btw @MiryangJung @softmarshmallow, thank you so much for taking the time to write docs. Just to give you an update - one of our team members is testing the flow and once they are done we should be good to go.
> 
> I know it's probably tiring to wait but we seek your patience - we think this is a good feature to have but we want to ensure that there are no issues with the rollout

fully understand. Thank you!

**Comment by ssudaraka** [comment]
Just an update: I was able to successfully test `supabase.auth.signInWithIdToken()` and it worked for me! However, I'm not sure if there are some edge cases I should test. I will sync with @J0 tomorrow on this. Sorry it took some time to test this.

**Review by J0 — APPROVED** [review]
Tested this locally as well and seems fine to me

**Comment by J0** [comment]
Note: test this again before next deploy or ping @J0

**Comment by ganer9r** [comment]
I have reviewed the code, but when will it be applied to the Supabase service? 
I am still encountering the following error: AuthApiError: Custom OIDC provider 'kakao' not allowed.

**Comment by kangmingtay** [comment]
@ganer9r merging the PR does not indicate that it's applied to the Supabase service, we need some time to roll it our to the platform which will take at least a week or so
